modules/ROOT/pages/anypoint-mq/3.x/anypoint-mq-listener.adoc

```diff
@@ -0,0 +1,347 @@
+
+To reference a circuit breaker:
+
+. Select the `subscriber` element in the canvas.
+. Click the *Advanced* tab.
+. Select *Circuit Breaker > Global Reference*, and then choose one from the list.
+
+image::anypoint-mq/3.x/amq-3x-cb-global-ref.png[]
+
+
+==== Private Circuit Breaker
+
+You declare a Private Circuit Breaker internally on a single subscriber. This circuit declaration is  used only in the flow where the subscriber is declared, isolated from all the other circuits.
+
+To use this configuration:
+
+. Select the Subscriber in the canvas.
+. Click the *Advanced* tab.
+. Select *Circuit breaker > Edit Inline*, and then choose one from the list.
+
+image::anypoint-mq/3.x/amq-3x-cb-inline.png[]
+
+=== Circuit Breaker Examples
+
+==== Circuit Configuration for a Single Subscriber
+
+In this example, a single subscriber consumes messages from a queue and posts the messages to another service using its REST API. You can stop processing messages after 5 requests to the external service result in a timeout. Once processing stops, the subscriber waits for 30 seconds for the service to recover before retrying with a new message.
+
+For this example, you need one config with these circuit breaker parameters:
+
+[source,xml,linenums]
+----
+
+<anypoint-mq:config name="Anypoint_MQ_Config">
+    <anypoint-mq:connection url="${providerUrl}"
+    clientId="${clientId}"
+    clientSecret="${clientSecret}"/>
+</anypoint-mq:config>
+
+<flow name="subscribe">
+   <anypoint-mq:subscriber config-ref="Anypoint_MQ_Config"
+      destination="${subscribedQueue}">
+        <anypoint-mq:circuit-breaker
+          onErrorTypes="HTTP:TIMEOUT"   <!-- (1) -->
+          errorsThreshold="5"           <!-- (2) -->
+          tripTimeout="30"              <!-- (3) -->
+          tripTimeoutUnit="SECONDS"/>
+   </anypoint-mq:subscriber>
+    <http:request config-ref="RequesterConfig"
+       path="/external" method="POST"/> <!-- (4) -->
+</flow>
+
+----
+[%noheader,cols="8,92,frame="none"]
+|===
+| *(1)* | Configures the error types to trip the circuit. When an error occurs for an `errorsThreshold` amount of times, polling stops.
+| *(2)* | Sets the threshold for how many consequent messages must occur to consider the circuit to be in a failure state.
+| *(3)* | Specifies how long to wait before resuming new message polling after the circuit breaker trips because the `errorsThreshold` is reached.
+| *(4)* | Defines the operation to throw the error expected by the `onErrorTypes` parameters.
+|===
+
+The circuit breaker ignores all errors that aren't listed in the `onErrorTypes` parameter. In this example, the circuit breaker ignores errors such as `HTTP:BAD_REQUEST`.
+
+
+====  Share a Circuit from Different Queues
+
+In many cases, there's a single common service between the processing of messages from different queues. This example configures the `circuitName` parameter to bind both subscribers to a single circuit:
```

Comment:
To avoid using "there is" as the subject of the sentence, how about "A single common service often processes messages from different queues."